packages/digitalriver-integration/src/digitalriver-payment-strategy.ts

```diff
@@ -160,7 +161,9 @@ export default class DigitalRiverPaymentStrategy implements PaymentStrategy {
                     throw error;
                 }
 
-                const confirm = await this.authenticateSource(error.body.provider_data);
+                const confirm =
+                    isRequestError(error) &&
+                    (await this.authenticateSource(error.body.provider_data));
```

Comment:
Should this code: https://github.com/bigcommerce/checkout-sdk-js/pull/2650/files#diff-d3f011ca802d01c870690db6cc81bd136939e3739490b1e2b7152f9825b77417R385 be called even if the `error` is not `requestError`?

Reply:
I think L160 elimates the possbility of `error`.
`isRequestError(error)` here just ensures the existence of `body` to make TS happy.